crates/stark_hash/src/felt.rs

```diff
@@ -57,6 +57,25 @@ impl<T> Dummy<T> for Felt {
     }
 }
 
+impl From<cairo_vm::felt::Felt252> for Felt {
+    fn from(value: cairo_vm::felt::Felt252) -> Self {
+        // unwrap() is safe here because Felt252 is guaranteed to not exceed the group size P
+        Self::from_be_slice(&value.to_be_bytes()).unwrap()
+    }
+}
+
+impl From<Felt> for cairo_vm::felt::Felt252 {
+    fn from(val: Felt) -> Self {
+        cairo_vm::felt::Felt252::from_bytes_be(val.as_be_bytes())
+    }
+}
+
+impl From<&Felt> for cairo_vm::felt::Felt252 {
+    fn from(val: &Felt) -> Self {
+        cairo_vm::felt::Felt252::from_bytes_be(val.as_be_bytes())
+    }
+}
+
```

Comment:
I don't really like the fact that we have to do this in this -- otherwise unrelated -- module, and add `cairo-vm` to the dependencies (and with the correct version number matching the one in starknet_in_rust, otherwise it won't work).

Reply:
Something we can and probably should do to guard against this, is create a `Felt` wrapper in common. Which hides the inner felt type we use. This would let us swop out the felt type without refactoring the world.

What we can do to "fix" this particular instance is to create a local wrapper at the usage site - or a local extension trait? (I haven't reviewed this yet, so maybe that's not possible). Similar to how the storage trait works.

Reply:
Hmm, maybe a trait could work. We're using these for a _lot_ of conversion to/from the starknet_rs felt type via `Into`. Could create and use a `FromFelt` / `IntoFelt` trait pair that we could implement in the `starknet_rs` code.

Reply:
Could also be more specific if you want. i.e. closer to `Into::into` where you can specify the target as well. So then you can map from `ContractAddress` directly to `ContractAddress` etc.

Reply:
Yes, we can do that too. starknet_in_rust is a lot _less_ typed than pathfinder though -- there are a lot of raw Felt252 values passed all over the code.

Reply:
Whatever's easiest then